gnomad_qc/v3/sample_qc/sample_qc.py

```diff
@@ -492,40 +1026,77 @@ def main(args):
         ).cols().select('inbreeding')
         inbreeding_ht.write(sample_inbreeding.path, overwrite=args.overwrite)
 
+    if args.calculate_clinvar:
+        mt = get_gnomad_v3_mt(
+            split=True,
+            key_by_locus_and_alleles=True,
+            remove_hard_filtered_samples=True
+        )
+        clinvar_ht = clinvar.ht()
+        mt = mt.filter_rows(hl.is_defined(clinvar_ht[mt.row_key]))
+        mt = mt.checkpoint("gs://gnomad-tmp/clinvar_variants.mt", overwrite=True)
+        mt = mt.annotate_rows(clinvar_path=hl.is_defined(filter_to_clinvar_pathogenic(clinvar_ht)[mt.row_key]))
+        clinvar_sample_ht = mt.select_cols(
+            n_clinvar=hl.agg.count_where(mt.GT.is_non_ref()),
+            n_clinvar_path=hl.agg.count_where(mt.GT.is_non_ref() & mt.clinvar_path),
+        ).cols()
+        clinvar_sample_ht.write(sample_clinvar_count.path, overwrite=args.overwrite)
+
     if args.apply_stratified_filters:
+        filtering_qc_metrics = args.filtering_qc_metrics.split(",")
+        sample_qc_ht = get_sample_qc('bi_allelic').ht()
+
+        if "inbreeding" in filtering_qc_metrics:
+            inbreeding_ht = sample_inbreeding.ht()[sample_qc_ht.key]
+            sample_qc_ht = sample_qc_ht.annotate(sample_qc=sample_qc_ht.sample_qc.annotate(inbreeding=inbreeding_ht.inbreeding.f_stat))
+
         apply_stratified_filters(
-            args.filtering_qc_metrics.split(",")
+            sample_qc_ht,
+            filtering_qc_metrics,
         ).write(stratified_metrics.path, overwrite=args.overwrite)
 
     if args.apply_regressed_filters:
+        n_pcs = args.regress_n_pcs
+        filtering_qc_metrics = args.filtering_qc_metrics.split(",")
+        sample_qc_ht = get_sample_qc('bi_allelic').ht()
+
+        if "inbreeding" in filtering_qc_metrics:
+            inbreeding_ht = sample_inbreeding.ht()[sample_qc_ht.key]
+            sample_qc_ht = sample_qc_ht.annotate(sample_qc=sample_qc_ht.sample_qc.annotate(inbreeding=inbreeding_ht.inbreeding.f_stat))
+
```

Comment:
L1046-L1052 and L1060-L1066 are identical. Does it make sense to move these out of the if statements and put it above the first one? Or Is it the case that sometimes we never want to apply the stratified or regressed filters so that code shouldn't be executed? This way, the statement within the if.args can be just the apply filter methods (plus the n_pcs  for regressed filters).  Alternatively, maybe a new arg along the lines of "apply_metric_filters" and nested within that could be the other if statements?

Reply:
For v3.1, I only really looked at the regressed filters (though I do think I ran both just to have them). So maybe just
```
    if args.apply_stratified_filters or args.apply_regressed_filters:
        filtering_qc_metrics = args.filtering_qc_metrics.split(",")
        sample_qc_ht = get_sample_qc('bi_allelic').ht()

        if "inbreeding" in filtering_qc_metrics:
            inbreeding_ht = sample_inbreeding.ht()[sample_qc_ht.key]
            sample_qc_ht = sample_qc_ht.annotate(sample_qc=sample_qc_ht.sample_qc.annotate(inbreeding=inbreeding_ht.inbreeding.f_stat))

        if args.apply_regressed_filters:
            n_pcs = args.regress_n_pcs
            apply_regressed_filters(
                sample_qc_ht,
                filtering_qc_metrics,
                args.include_unreleasable_samples,
                n_pcs,
            ).write(regressed_metrics.path, overwrite=args.overwrite)
        else:
            apply_stratified_filters(
                sample_qc_ht,
                filtering_qc_metrics,
            ).write(stratified_metrics.path, overwrite=args.overwrite)
```